### PR TITLE
refactor: generalize settings file generation for the maven plugin

### PR DESCRIPTION
**Pull request by bepri**
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you added an entry to the changelog (`docs/reference/changelog.rst`)?

-----

Upstreams the generation of the Maven settings file to a Maven utilities file in preparation for the maven-use plugin. I followed the lead of how the settings file was generated in [this branch](https://github.com/canonical/craft-parts/blob/4cc452ee31c99f79d96635d2bcc2ed765e7de12b/craft_parts/plugins/_maven_util.py).

The change in `craft_parts/__init__.py` was to fix a circular import error when directly importing `craft_parts.infos`.

CRAFT-4569

### CODE REVIEW

File: craft_parts/plugins/maven_plugin.py [hunk 1]
```diff
@@ -16,16 +16,13 @@
...
 import re
-import xml.etree.ElementTree as ET
-from pathlib import Path
 from typing import Literal, cast
-from urllib.parse import urlparse
 
 from overrides import override
 
 from craft_parts import errors
+from craft_parts.plugins._maven_util import create_maven_settings
```
Comment (tigarmo) [inline craft_parts/plugins/maven_plugin.py:25]: maybe `_maven_util` and `_maven_xml` should be in a new `maven` package inside `plugins`?

**Review by tigarmo — CHANGES_REQUESTED** [review]

File: tests/unit/plugins/test_maven_plugin.py [hunk 2]
```diff
@@ -224,20 +237,22 @@ def test_settings_proxy(
...
           </proxies>
         </settings>
         """
     )
 
     env_dict = {
-        f"{protocol}_proxy": "http://my-proxy-host:3128",
-        f"{protocol}_PROXY": "http://my-proxy-host:3128",
+        f"{protocol}_proxy": "http://username:hunter7@my-proxy-host:3128",
+        f"{protocol}_PROXY": "http://username:hunter7@my-proxy-host:3128",
```
[the remark this answers is not in the document]
Comment (bepri) [reply tests/unit/plugins/test_maven_plugin.py:246]: correct, there weren't any tests for it yet

**Review by tigarmo — CHANGES_REQUESTED** [review]

**Review by tigarmo — APPROVED** [review]
thanks!

File: craft_parts/plugins/maven_plugin.py
Comment (lengau) [inline craft_parts/plugins/maven_plugin.py:1]: Copyright needs updating

File: tests/unit/plugins/test_maven_plugin.py [hunk 3]
```diff
@@ -224,20 +226,22 @@ def test_settings_proxy(
           <proxies>
             <proxy>
               <id>{expected_protocol}_proxy</id>
-              <active>true</active>
               <protocol>{expected_protocol}</protocol>
               <host>my-proxy-host</host>
               <port>3128</port>
               <nonProxyHosts>{non_proxy_hosts}</nonProxyHosts>
+              <username>username</username>
+              <password>hunter7</password>
```
Comment (cmatsuoka) [inline tests/unit/plugins/test_maven_plugin.py:234]: We can also have a test with empty credentials to make sure it's generating the right output.
